avm/res/insights/data-collection-rule/main.bicep

```
@@ -152,15 +152,19 @@ output resourceId string = dataCollectionRuleProperties.kind == 'All' ? dataColl
 output resourceGroupName string = resourceGroup().name
 
 @description('The location the resource was deployed into.')
-output location string = dataCollectionRuleProperties.kind == 'All'
-  ? dataCollectionRuleAll.location
-  : dataCollectionRule.location
+output location string = dataCollectionRuleProperties.kind == 'All' ? dataCollectionRuleAll.location : dataCollectionRule.location
 
 @description('The principal ID of the system assigned identity.')
 output systemAssignedMIPrincipalId string? = dataCollectionRuleProperties.kind == 'All'
   ? dataCollectionRuleAll.?identity.?principalId
   : dataCollectionRule.?identity.?principalId
 
+@description('The endpoints of the dataCollectionRule, if created.')
+output endpoints object? = dataCollectionRuleProperties.kind == 'All' ? dataCollectionRuleAll.properties.?endpoints : dataCollectionRule.properties.?endpoints
```

Hey @ckane,
I'll take this one for @krbar as he's out-of-office and you opened this PR  a while ago.

A suggestion if I amy for this one though:
```suggestion
output endpoints resourceOutput<'Microsoft.Insights/dataCollectionRules@2023-03-11'>.properties.endpoints? = dataCollectionRuleProperties.kind == 'All'
  ? dataCollectionRuleAll!.properties.?endpoints
  : dataCollectionRule!.properties.?endpoints
```

Typed outputs are great as they allow you to 'discover' the output when using the module (so you don't need to know the properties by heart)